### ISSUE TITLE
Memory leak when uploading lots of images?

### ISSUE THREAD

**Issue by lukebennett88** [cloudinary-devs/gatsby-transformer-cloudinary#11]
I'm trying to replace images served from Netlify with ones served from Cloudinary on an existing site which contains quite a lot of images, however when I run `gatsby develop` or `gatsby build` I get the following error:

```
success onPreBootstrap - 0.008s

 ERROR 

undefined undefined



  Error: 

not finished source and transform nodes - 12.517s

 ERROR 

Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in StoreStateProvider
    in App
```

Looking in my uploads folder in Cloudinary there seem to be a few more images uploaded each time I run `gatsby develop` or `gatsby build` so it looks like it's during the upload process that the memory leak gets caught and the process is cancelled.

**Comment by Chuloo** [comment]
Hi, @lukebennett88 what's the best way to reproduce your issue? I can't seem to reproduce on my end.

**Comment by rheinardkorf** [comment]
I believe this is related to #17 . Also experienced this. When using large images I experience what happens in #17, when smaller but more images then I get the same result as @lukebennett88 .

To reproduce my experience, use `gatsby-source-filesystem` to source approx 20 images.  I am pointing it to a local path `content` and images are in a nested folder `content/images`. This is enough to cause the leak issue.

FYI, when using `gatsby build` I often have more success than using `gatsby develop`.

**Comment by jlengstorf** [comment]
maybe uploading in batches would help? I don't have a specific approach that I know would work, but something like running a check on file size and limiting at XMB at a time?

**Comment by Chuloo** [comment]
This issue is fixed by #21, and will be closed. Please re-open if you face the same issue again.
